### ISSUE TITLE
[P1] Ability to switch between using tool call API and code executor for executing skills

### ISSUE THREAD

**Issue by victordibia** [microsoft/autogen#3058]
AutoGen Studio currently supports only adding skills/tools to the LLM prompt.
However, many users want to call skills/tools via the function calling capabilities of some LLMs.
This issue explores supporting toggling between adding skill to prompt and function calling approach to using tools. 

@qingyun-wu

**Comment by SailorJoe6** [comment]
I have this working in my local development environment, but it needs some polish.  Once my other PR is merged, I'm happy to polish this up and submit another PR @victordibia

**Comment by SailorJoe6** [comment]
@victordibia Hoping you've got some bandwidth to have a glance at my PR #3491

**Comment by victordibia** [comment]
With the new updates to AGS, this is now addressed.
The short story is that  the new version standardizes on a generic tool interface - a code executor can be a tool that is called using the tool calling capabilities of the model, similar to any function or the host of tool types supported .

- AutoGen Studio documentation - https://microsoft.github.io/autogen/dev/user-guide/autogenstudio-user-guide/index.html 
- To learn about how code executors work ...  https://microsoft.github.io/autogen/dev/user-guide/core-user-guide/design-patterns/code-execution-groupchat.html 
- How to convert any agentchat python code and use it in autogen studio - https://microsoft.github.io/autogen/dev/user-guide/autogenstudio-user-guide/usage.html#declarative-specification-of-componenents
